src/Json_encode.mli

```diff
@@ -6,6 +6,9 @@ type 'a encoder = 'a -> Js.Json.t
 external null : Js.Json.t = "" [@@bs.val]
 (** [null] is the singleton null JSON value *)
 
+external undefined : Js.Json.t = "" [@@bs.val]
+(** [undefined] values will not show up in the json output *)
```

Comment:
This isn't actually true. A field having a value of `undefined` is not equivalent to the field not being defined, although it's not always possible to distinguish the two. There are APIs that treat them differently however, such as `fetch`.

It might be better not not include this and `optional` to avoid confusion.

Reply:
Its not true in javascript, but it is true for JSON.